### PR TITLE
✨ Add Auth0 Routes and Configuration

### PR DESCRIPTION
**Pull request by jasonBirchall**
Connects to https://github.com/ministryofjustice/operations-engineering/issues/4839

### CODE REVIEW

File: app/main/routes/auth.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@auth_route.route("/logout", methods=["GET", "POST"])
+def logout():
+    session.clear()
+    return auth0_service.logout(url_for("main.index", _external=True))
+
+
+@auth_route.route("/callback", methods=["GET", "POST"])
+def callback():
+    session["user"] = auth0_service.get_access_token()
+    return redirect("/join/send-invitation")
```
Comment (connormaglynn) [inline app/main/routes/auth.py:34]: Need to update the callback redirect ♻️

File: app/main/config/app_config.py [hunk 2]
```diff
@@ -7,6 +7,11 @@ def __get_env_var(name: str) -> str | None:
 
 
 app_config = SimpleNamespace(
+    auth0=SimpleNamespace(
+        domain=__get_env_var("AUTH0_DOMAIN"),
```
Comment (connormaglynn) [inline app/main/config/app_config.py:11]: We will also need to inject these in the Deployment GitHub Workflow, and map the values in the Helm charts 🗺️

File: .github/workflows/cicd-deploy-to-dev.yml [hunk 3]
```diff
@@ -15,6 +15,8 @@ env:
   ECR_REPOSITORY: ${{ vars.DEV_ECR_REPOSITORY }}
   ADMIN_GITHUB_TOKEN: ${{ secrets.ADMIN_GITHUB_TOKEN }}
   ADMIN_SLACK_TOKEN: ${{ secrets.ADMIN_SLACK_TOKEN }}
+  AUTH0_CLIENT_ID: ${{ secrets.DEV_AUTH0_CLIENT_ID }}
```
Comment (connormaglynn) [inline .github/workflows/cicd-deploy-to-dev.yml:18]: These will also need to be added as parameters in the helm command further down 👇 Line 70 😁

Comment (jasonBirchall) [reply .github/workflows/cicd-deploy-to-dev.yml:18]: Excellent. Done. Thanks

**Review by connormaglynn — APPROVED** [review]
🚀 Looks Good To Me!

**Pull request merged**